sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineSize.java

```diff
@@ -22,4 +22,14 @@
 
     /** @return the maximum number of data disks allowed by a VM size */
     int maxDataDiskCount();
+
+    /**
+     * The virtual machine size type if the sku describes sku for virtual machine resource type.
+     *
+     * <p>The size can be used for {@link VirtualMachine.DefinitionStages.WithVMSize#withSize(VirtualMachineSizeTypes)}
+     * and {@link VirtualMachine.Update#withSize(VirtualMachineSizeTypes)}.
+     *
+     * @return the virtual machine size type
+     */
+    VirtualMachineSizeTypes virtualMachineSizeType();
```

Comment:
Will `type()` be enough? Since it's under the context of `VirtualMachineSize`.

Reply:
I am afraid that `type` might get it confused with Resource.type (which is almost in every resource)